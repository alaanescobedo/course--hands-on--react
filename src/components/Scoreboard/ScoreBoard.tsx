import styled from "@emotion/styled";
import { FC } from "react";
import { Counter } from "./Counter";
import { Level } from "./Level";
import { Reset } from "./Reset";

export interface ScoreboardProps {
  /**
   * Timer
   */
  time: string;
  /**
   * Possible game scenarios
   */
  levels: string[];
  /**
   * Action handler when the GameReset button is clicked
   */
  mines: string;
  /**
   * Bombs in the field
   */
  onReset: () => void;
}

export const Scoreboard: FC<ScoreboardProps> = ({
  time,
  levels,
  mines,
  onReset,
}) => (
  <Wrapper>
    <Counter>{time}</Counter>
    <Level>{levels}</Level>
    <Reset onReset={onReset}/>
    <Counter>{mines}</Counter>
  </Wrapper>
);

const Wrapper = styled.div`
  display: flex;
  width: max-content;
  padding-bottom: 2vw;
  justify-content: space-between;
`;
