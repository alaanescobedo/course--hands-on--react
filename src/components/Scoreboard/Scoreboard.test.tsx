import { render } from '@testing-library/react';

import { Scoreboard } from './ScoreBoard';

it('Scoreboard renders correctly', () => {
  const { asFragment } = render(
    <Scoreboard
      time="000"
      levels={['beginner', 'intermediate', 'expert']}
      mines="010"
      onReset={() => null}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});