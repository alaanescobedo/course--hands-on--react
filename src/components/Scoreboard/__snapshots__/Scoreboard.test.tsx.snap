// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1gexv9m"
  >
    <div
      class="css-q796n5"
    >
      000
    </div>
    <select
      class="css-oyqqd"
    >
      <option
        class="css-zcmdxr"
        value="beginner"
      >
        beginner
      </option>
      <option
        class="css-zcmdxr"
        value="intermediate"
      >
        intermediate
      </option>
      <option
        class="css-zcmdxr"
        value="expert"
      >
        expert
      </option>
    </select>
    <button
      class="css-15ja8qh"
    >
      ðŸ™‚
    </button>
    <div
      class="css-q796n5"
    >
      010
    </div>
  </div>
</DocumentFragment>
`;
